src/autotester/server/utils/tests/path_management_test.py

```diff
@@ -0,0 +1,23 @@
+from src.autotester.server.utils.path_management import current_directory, add_path
+import tempfile
+import os
+import sys
+
+
+def test_current_directory():
+    tmp_dir = tempfile.gettempdir()
+    curr_dir = os.getcwd()
+    with current_directory(tmp_dir):
+        temp_dir = os.getcwd()
+    assert temp_dir == tmp_dir
+    assert os.getcwd() == curr_dir
+
+
+def test_add_path():
```

Comment:
These tests look great.

Can you add another that checks: 

- if a path is already in `sys.path` and then the same path is added again, then make sure it is added properly and that only the one path is removed (not both) at the end